lib/Sema/CSSimplify.cpp

```diff
@@ -13584,6 +13584,16 @@ ConstraintSystem::SolutionKind ConstraintSystem::simplifyShapeOfConstraint(
     return SolutionKind::Solved;
   }
 
+  if (isSingleUnlabeledPackExpansionTuple(packTy)) {
+    auto *packVar =
+        createTypeVariable(getConstraintLocator(locator), TVO_CanBindToPack);
```

Comment:
I think the locator for everything here except to the new `ShapeOf` constraint needs to be `locator` without its last element (which is `PackShape`).